app/models/session.rb

```diff
@@ -57,16 +72,26 @@ def device_info
   end
 
   def as_client_payload
+    # TODO: this method should only be called on a newly created instance or after
+    # calling renew method. The reason is that access_token and refresh_token are 
+    # just class attributes and are not really persisted to database.
     {
-      access_token: access_token,
-      refresh_token: refresh_token,
+      access_token: "1:#{uuid}:#{access_token}",
```

Comment:
Version should be extracted into a variable so that it's self evident what it represents.